pypsa/optimization/constraints.py

```diff
@@ -105,16 +154,391 @@ def define_operational_constraints_for_extendables(
+        mask = (must_stay_down >= timesteps) & initially_down
+        name = f"{c}-com-mod-status-min_down_time_must_stay_up" + name_func
+        mask = mask & active if active is not None else mask
+        n.model.add_constraints(status, "=", 0, name=name, mask=mask)
+
+    # linearized approximation because committable can partly start up and shut down
+    cost_equal = all(
+        n.df(c).loc[com_i, "start_up_cost"] == n.df(c).loc[com_i, "shut_down_cost"]
+    )
+    # only valid additional constraints if start up costs equal to shut down costs
+    if n._linearized_uc and not cost_equal:
+        logger.warning(
+            "The linear relaxation of the unit commitment cannot be "
+            "tightened since the start up costs are not equal to the "
+            "shut down costs. Proceed with the linear relaxation "
+            "without the tightening by additional constraints. "
+            "This might result in a longer solving time."
+        )
+    if n._linearized_uc and cost_equal:
+        # dispatch limit for partly start up/shut down for t-1
+        lhs = (
+            p.shift(snapshot=1)
+            - ramp_shut_down * status.shift(snapshot=1)
+            - (upper_p - ramp_shut_down) * (status - start_up)
+        )
+        lhs = lhs.sel(snapshot=sns[1:])
+        n.model.add_constraints(
+            lhs, "<=", 0, name=f"{c}-com-mod-p-before" + name_func, mask=active
+        )
+
+        # dispatch limit for partly start up/shut down for t
+        lhs = p - upper_p * status + (upper_p - ramp_start_up) * start_up
+        lhs = lhs.sel(snapshot=sns[1:])
+        n.model.add_constraints(
+            lhs, "<=", 0, name=f"{c}-com-mod-p-current" + name_func, mask=active
+        )
+
+        # ramp up if committable is only partly active and some capacity is starting up
+        lhs = (
+            p
+            - p.shift(snapshot=1)
+            - (lower_p + ramp_up_limit) * status
+            + lower_p * status.shift(snapshot=1)
+            + (lower_p + ramp_up_limit - ramp_start_up) * start_up
+        )
+        lhs = lhs.sel(snapshot=sns[1:])
+        n.model.add_constraints(
+            lhs, "<=", 0, name=f"{c}-com-mod-partly-start-up" + name_func, mask=active
+        )
+
+        # ramp down if committable is only partly active and some capacity is shutting up
+        lhs = (
+            p.shift(snapshot=1)
+            - p
+            - ramp_shut_down * status.shift(snapshot=1)
+            + (ramp_shut_down - ramp_down_limit) * status
+            - (lower_p + ramp_down_limit - ramp_shut_down) * start_up
+        )
+        lhs = lhs.sel(snapshot=sns[1:])
+        n.model.add_constraints(
+            lhs, "<=", 0, name=f"{c}-com-mod-partly-shut-down" + name_func, mask=active
+        )
+
 
-def define_operational_constraints_for_committables(n, sns, c):
+def define_operational_constraints_for_committables_non_modular_non_extendables(
```

Comment:
As mentioned, avoid renaming functions and constraints unless strictly necessary.
I'd avoid that for the moment for backward compatibility

Reply:
I know and I agree. As explained before, it seemed necessary to me, since these names are based on the idea that ext and comm are not compatible. However, tell me if you do not agree that I will change.